### PR TITLE
Add MCP Abilities API integration

### PR DESCRIPTION
**Pull request by chrisfromthelc**
## Summary

- Extract `Issue_Detector` from `AI_Formatter` for reuse by MCP abilities and other consumers
- Add `Abilities_Provider` class registering 6 abilities with the WordPress Abilities API (WP 6.9+):
  - `get-issues` — detected warnings/critical issues (readonly)
  - `get-report` — full system report in markdown or JSON (readonly)
  - `get-section` — single collector section by ID (readonly)
  - `get-error-log` — last N lines of PHP error log (readonly)
  - `get-debug-status` — WP_DEBUG/WP_DEBUG_LOG state (readonly)
  - `toggle-debug` — enable/disable debug logging (modifies wp-config.php)
- Wire into Plugin orchestrator via constructor injection
- All abilities include JSON Schema input/output, `manage_options` permission callback, and `show_in_rest` metadata
- Graceful degradation: no-op when Abilities API unavailable (WP < 6.9 or MCP adapter not installed)
- Update README with MCP integration section, example agent workflow, and updated architecture diagram

## Test Coverage

- `IssueDetectorTest` — 15 tests: detection behavior, issue structure, heuristic checks, edge cases
- `AbilitiesProviderTest` — 26 tests: hook registration, all 6 handler callbacks with mocks, permissions, graceful degradation
- Existing `FormattersTest` — regression coverage for `Issue_Detector` extraction (25 tests unchanged)
- **Total: 220 tests, 2010 assertions — all passing**

## CI Checks

- PHPCS: clean
- PHPStan: no errors
- PHPUnit: 220/220 pass
- Rector dry-run: no new changes (existing `@var` suggestion predates this PR)

## Test plan

- [ ] CI passes (PHPCS 8.1-8.4, PHPStan, PHPUnit 8.1-8.4, Rector dry-run)
- [ ] Activate MCP Adapter plugin on WP 6.9+
- [ ] Verify `wp-json/wp-abilities/v1/abilities` lists all 6 `wp-system-report/*` abilities
- [ ] Call `get-issues` via MCP client and confirm structured response
- [ ] Call `get-report` with both `markdown` and `json` formats
- [ ] Call `get-section` with valid ID and invalid ID (confirm error + available_sections)
- [ ] Call `get-error-log` and confirm redacted log lines
- [ ] Call `get-debug-status` and confirm state matches wp-config.php
- [ ] Call `toggle-debug` and verify wp-config.php changes
- [ ] Verify existing admin page and REST API still work correctly

## Summary by Sourcery

Integrate the plugin with the WordPress MCP Abilities API to expose system-reporting and debugging capabilities to AI agents, while extracting reusable issue-detection logic and enriching responses with environment-aware guidance.

New Features:
- Add an Abilities_Provider that registers seven MCP abilities for querying issues, fetching reports and sections, reading error logs, inspecting debug status, toggling debug logging, and retrieving agent context via the WordPress Abilities API.
- Introduce an Agent_Guidance helper that exposes environment-aware rules, thresholds, PHP lifecycle data, WooCommerce-specific guidance, and per-ability hints for AI agents.
- Expose a reusable Issue_Detector service for deriving issues from report data, shared between the AI formatter and MCP abilities.

Enhancements:
- Wire the Abilities_Provider into the main plugin so abilities are registered when the MCP Abilities API is available, without affecting behavior on older WordPress installs.
- Update the AI formatter to delegate issue detection to the new Issue_Detector, preserving existing formatting behavior.
- Extend the README with MCP integration details, example agent workflows, and an updated architecture overview including the new classes.

Tests:
- Add comprehensive unit tests for the Abilities_Provider covering hook registration, all handlers, permissions, and environment enrichment.
- Add dedicated tests for Issue_Detector behavior and heuristics, and for Agent_Guidance to validate environment context, rules, thresholds, lifecycle data, and WooCommerce guidance.

### CODE REVIEW

**Comment by chrisfromthelc** [comment]
@sourcery-ai review

**Comment by chrisfromthelc** [comment]
@sourcery-ai review

**Comment by chrisfromthelc** [comment]
@sourcery-ai review

File: includes/class-abilities-provider.php [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+		$issues      = $detector->detect( $report_data );
+
+		/** This filter is documented in includes/formatters/class-ai-formatter.php */
+		$issues = apply_filters( 'wp_system_report_ai_issues', $issues, $report_data );
+
+		$critical_count = 0;
+		$warning_count  = 0;
+
+		foreach ( $issues as $issue ) {
+			if ( 'critical' === $issue['severity'] ) {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php:507]: Fixed.

File: includes/class-abilities-provider.php [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+				'label'               => __( 'Get System Issues', 'wp-system-report' ),
+				'description'         => __( 'Returns detected warnings and critical issues from the system report. Use this to quickly identify what needs attention.', 'wp-system-report' ),
+				'category'            => 'wp-system-report',
+				'execute_callback'    => array( $this, 'handle_get_issues' ),
+				'permission_callback' => array( $this, 'check_manage_options' ),
+				'input_schema'        => array(
+					'type'       => 'object',
+					'properties' => new \stdClass(),
+				),
+				'output_schema'       => array(
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php:174]: Fixed.

File: includes/class-abilities-provider.php [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+	 */
+	public function handle_get_section( array $input ): array {
+		$section_id = isset( $input['section'] ) ? $input['section'] : '';
+		$section    = $this->report_generator->generate_section( $section_id );
+
+		if ( null === $section ) {
+			$collectors         = $this->report_generator->get_collectors();
+			$available_sections = array_keys( $collectors );
+
+			return array(
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php]: Fixed.

File: includes/class-issue-detector.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	}
+
+	/**
+	 * Build a descriptive issue message from a field.
+	 *
+	 * @param array $field Field data.
+	 * @return string Issue description.
+	 */
+	private function build_issue_description( $field ): string {
+		$desc = 'Current value: ' . $field['value'] . '.';
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php]: Fixed.

File: includes/class-issue-detector.php [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+		return $issues;
+	}
+
+	/**
+	 * Build a descriptive issue message from a field.
+	 *
+	 * @param array $field Field data.
+	 * @return string Issue description.
+	 */
+	private function build_issue_description( $field ): string {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php]: Fixed.

File: includes/class-issue-detector.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php]: Fixed.

File: includes/class-issue-detector.php [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+	 * @param array $report_data Full report data.
+	 * @return array Additional issues found.
+	 */
+	private function run_heuristic_checks( array $report_data ): array {
+		$issues = array();
+
+		// Check PHP version EOL.
+		$php_version = phpversion();
+		if ( version_compare( $php_version, '8.1', '<' ) ) {
+			$issues[] = array(
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:106]: Fixed.

File: includes/class-issue-detector.php [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+	 * Detect issues from the report data.
+	 *
+	 * Scans all fields for warnings and critical statuses, and runs
+	 * additional heuristic checks for common problems.
+	 *
+	 * @param array $report_data Full report data keyed by collector ID.
+	 * @return array Array of issue arrays with 'severity', 'title', 'description' keys.
+	 */
+	public function detect( array $report_data ): array {
+		$issues = array();
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:30]: Fixed.

File: includes/class-issue-detector.php [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+	 *
+	 * @param array $report_data Full report data.
+	 * @return array Additional issues found.
+	 */
+	private function run_heuristic_checks( array $report_data ): array {
+		$issues = array();
+
+		// Check PHP version EOL.
+		$php_version = phpversion();
+		if ( version_compare( $php_version, '8.1', '<' ) ) {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:105]: Fixed.

File: includes/class-issue-detector.php [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+			$issues[] = array(
+				'severity'    => 'critical',
+				'title'       => 'PHP version ' . $php_version . ' is end-of-life',
+				'description' => 'This PHP version no longer receives security updates. Upgrade to PHP 8.1 or newer.',
+			);
+		}
+
+		// Check if object cache is in use when many plugins are active.
+		if ( ! wp_using_ext_object_cache() ) {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:118]: Fixed.

File: includes/class-issue-detector.php [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+				);
+			}
+		}
+
+		// Autoloaded options size is already reported by the collector with field-level
+		// status. The detect() method captures it automatically — no duplicate needed.
+
+		// Check for non-InnoDB tables from existing report data (Database section).
+		$database_section = $this->find_section_by_id( $report_data, 'database' );
+		if ( $database_section ) {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php]: Fixed.

File: includes/class-abilities-provider.php [hunk 11]
```diff
@@ -0,0 +1,551 @@
...
+		$critical_count = 0;
+		$warning_count  = 0;
+
+		foreach ( $issues as $issue ) {
+			if ( 'critical' === $issue['severity'] ) {
+				++$critical_count;
+			} else {
+				++$warning_count;
+			}
+		}
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php:512]: Fixed.

File: includes/class-issue-detector.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php]: Fixed.

File: includes/class-issue-detector.php [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+		}
+
+		// Autoloaded options size is already reported by the collector with field-level
+		// status. The detect() method captures it automatically — no duplicate needed.
+
+		// Check for non-InnoDB tables from existing report data (Database section).
+		$database_section = $this->find_section_by_id( $report_data, 'database' );
+		if ( $database_section && ! empty( $database_section['fields'] ) && is_array( $database_section['fields'] ) ) {
+			$non_innodb_count = 0;
+			foreach ( $database_section['fields'] as $field ) {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:140]: Fixed.

File: includes/class-abilities-provider.php [hunk 13]
```diff
@@ -0,0 +1,590 @@
...
+		wp_register_ability(
+			'wp-system-report/get-issues',
+			array(
+				'label'               => __( 'Get System Issues', 'wp-system-report' ),
+				'description'         => __( 'Returns detected warnings and critical issues from the system report. Use this to quickly identify what needs attention.', 'wp-system-report' ),
+				'category'            => 'wp-system-report',
+				'execute_callback'    => array( $this, 'handle_get_issues' ),
+				'permission_callback' => array( $this, 'check_manage_options' ),
+				'input_schema'        => array(
+					'type'       => 'object',
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php]: Fixed.

File: includes/class-abilities-provider.php [hunk 14]
```diff
@@ -0,0 +1,590 @@
...
+	 * Handle the get-report ability.
+	 *
+	 * @param array $input Ability input with optional 'format' key.
+	 * @return array Report data.
+	 */
+	public function handle_get_report( array $input ): array {
+		$format      = isset( $input['format'] ) ? $input['format'] : 'markdown';
+		$report_data = $this->report_generator->generate();
+
+		if ( 'json' === $format ) {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php]: Fixed.

File: includes/class-abilities-provider.php [hunk 15]
```diff
@@ -0,0 +1,551 @@
...
+ * Abilities API provider for the WordPress MCP Adapter.
+ *
+ * @package SystemReport
+ */
+
+namespace SystemReport;
+
+use SystemReport\Formatters\AI_Formatter;
+
+defined( 'ABSPATH' ) || exit;
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php:12]: Intentional — `defined( 'ABSPATH' ) || exit;` is this project's standard ABSPATH guard pattern (used in all existing files). `wp_die()` is not available before WordPress loads, so `exit` is correct here. See technical notes.

File: includes/class-issue-detector.php [hunk 16]
```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Issue detector.
+ *
+ * @package SystemReport
+ */
+
+namespace SystemReport;
+
+defined( 'ABSPATH' ) || exit;
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:10]: Intentional — `defined( 'ABSPATH' ) || exit;` is this project's standard ABSPATH guard pattern (used in all existing files). `wp_die()` is not available before WordPress loads, so `exit` is correct here. See technical notes.

File: includes/class-abilities-provider.php [hunk 17]
```diff
@@ -0,0 +1,557 @@
...
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php:12]: Intentional — `defined( 'ABSPATH' ) || exit;` is this project's standard ABSPATH guard pattern (used in all existing files). `wp_die()` is not available before WordPress loads, so `exit` is correct here. See technical notes.

File: includes/class-issue-detector.php [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:10]: Intentional — `defined( 'ABSPATH' ) || exit;` is this project's standard ABSPATH guard pattern (used in all existing files). `wp_die()` is not available before WordPress loads, so `exit` is correct here. See technical notes.

File: includes/class-abilities-provider.php [hunk 19]
```diff
@@ -0,0 +1,590 @@
...
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-abilities-provider.php:12]: Intentional — `defined( 'ABSPATH' ) || exit;` is this project's standard ABSPATH guard pattern (used in all existing files). `wp_die()` is not available before WordPress loads, so `exit` is correct here. See technical notes.

File: includes/class-issue-detector.php [hunk 20]
```diff
@@ -0,0 +1,177 @@
...
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-issue-detector.php:10]: Intentional — `defined( 'ABSPATH' ) || exit;` is this project's standard ABSPATH guard pattern (used in all existing files). `wp_die()` is not available before WordPress loads, so `exit` is correct here. See technical notes.

**Comment by chrisfromthelc** [comment]
@sourcery-ai review

File: includes/class-agent-guidance.php [hunk 21]
```diff
@@ -0,0 +1,471 @@
...
+	private static function detect_hpos_status(): string {
+		if ( ! self::is_woocommerce_active() ) {
+			return 'unknown';
+		}
+
+		if ( ! function_exists( 'wc_get_container' ) ) {
+			return 'unknown';
+		}
+
+		try {
```
[the remark this answers is not in the document]
Comment (chrisfromthelc) [reply includes/class-agent-guidance.php:434]: Fixed — changed `catch ( \\Exception $e )` to `catch ( \\Throwable $e )` so HPOS detection degrades gracefully for any throwable type, not just Exception subclasses.
